### ISSUE TITLE
Unable to install 21cmFast on Mac M1 processor [BUG]

### ISSUE THREAD

**Issue by chapmanemma** [21cmfast/21cmFAST#306]
I have tried to install 21cmFast on a new MacBook with an M1 chip. I have followed the instructions for installing using pip and while the installation completes, running "import py21cmfast as p21c" results in:

ImportError: dlopen(/Users/student/Documents/21cmFast/21cmFAST/src/py21cmfast/c_21cmfast.abi3.so, 0x0002): tried: '/Users/student/Documents/21cmFast/21cmFAST/src/py21cmfast/c_21cmfast.abi3.so' (mach-o file, but is an incompatible architecture (have 'arm64', need 'x86_64')), '/usr/local/lib/c_21cmfast.abi3.so' (no such file), '/usr/lib/c_21cmfast.abi3.so' (no such file)

**Comment by chapmanemma** [comment]
We have managed to install, but only using miniforge and pip.

**Comment by steven-murray** [comment]
Hi @chapmanemma -- I don't think any of us has had experience trying to install on the M1, sorry for the hassle. Any chance you could provide a little detail on the solution -- we could add it to the docs, or at least leave it here for the future :-)

**Comment by BradGreig** [comment]
Hi @chapmanemma @steven-murray, not that this is going to be overly useful but I do have 21cmFAST installed on an M1 Apple. I don't recall having any issues installing it though. However, I followed the installation instructions for developers from the docs. Not sure if there is much difference, but it does use pip.

Unfortunately I did not write down/nor do I remember the steps I followed.

I guess the primary issues will be making sure the versions of python and gcc are compatible with the M1 chip. I have gcc v11 and python3.10. FFTW would also need to be compiled under this version.

**Comment by chapmanemma** [comment]
cc’ing Luke who figured it out on this end. We were unable to install using the standard conda conda-forge install, but for some reason installing miniconda and then using the pip install command to install worked. It was as “simple” as that in the end. I’m away from my office so cannot reproduce all the errors at the moment but Luke may be able to add something.

On 17 Oct 2022, at 01:18, BradGreig ***@***.******@***.***>> wrote:



Hi @chapmanemma<https://github.com/chapmanemma> @steven-murray<https://github.com/steven-murray>, not that this is going to be overly useful but I do have 21cmFAST installed on an M1 Apple. I don't recall having any issues installing it though. However, I followed the installation instructions for developers from the docs. Not sure if there is much difference, but it does use pip.

Unfortunately I did not write down/nor do I remember the steps I followed.

I guess the primary issues will be making sure the versions of python and gcc are compatible with the M1 chip. I have gcc v11 and python3.10. FFTW would also need to be compiled under this version.
